src/backend/revoker/expiredInventory.ts

```diff
@@ -32,7 +36,13 @@ export async function fillExpiredInventory() {
   attestationsToRemove.push(...expiredSinceLastRun);
   attestationsToRemoveLater.splice(0, expiredSinceLastRun.length);
 
-  for await (const expiredAttestation of getExpiredAttestations()) {
+  const issuedBy = configuration.did;
+
+  if (issuedBy === 'pending') {
+    return;
+  }
+
+  for await (const expiredAttestation of queryExpiredAttestations(issuedBy)) {
```

Comment:
I just did not want to do this inside `queryExpiredAttestations`:

```
  const issuedBy = configuration.did;

  if (issuedBy === 'pending') {
    return;
  }
```

Reply:
Here is another possibility, but I'm not a fan:

```
export function queryExpiredAttestations() {
  const issuedBy = configuration.did;

  if (issuedBy === 'pending') {
    async function* voidGenerator() {
      return;
    }
    return voidGenerator;
  }
  ...
  }
```

With this:
```
function queryExpiredAttestations(): (() => AsyncGenerator<never, void, unknown>) | AsyncGenerator<AttestationInfo, void, unknown>
```

Reply:
Can see the alternative here:

`stash@{0}: On xw/use_indexer: refactor: parameter phobia`

Reply:
I will be against changing this.

Reply:
You are doing the right thing by returning when the DID is pending. 

However to avoid drilling the value of DID through several layers of functions I would rather tell TypeScript that we the developers know better. I guess if we put `configuration.did` directly in the query, TS won’t even complain. And the execution will only get to running the query when the `configuration.did` contains a DID.

Reply:
And you won’t have to change types

Reply:
Putting `configuration.did` directly in the query works. It would just return 0 matches if the DID is still **"pending"**. 

So it introduces an unnecessary **https** request.

Reply:
I don't see any benefits from this. 

The function with the parameter could potentially be use somewhere else. 

Why is it good to avoid parameter of functions? 
  
This function is even consuming this value,  it is de facto a parameter, why would it be good to hide it? 

I think that leaving it out as a parameter would make solving an error/unwanted behaviour easier. Through the error stack you could pinpoint that it is being caused by a missing DID instead of having to analyse the whole `src/backend/revoker/indexer/queryAttestations.ts` file.

Reply:
My point is that the HTTP request will never be fired if there’s no DID, that’s why I said:

> And the execution will only get to running the query when the configuration.did contains a DID

Every parameter increases the complexity ever so slightly, making the code a bit more difficult to analyze. If you follow through with the approach of making everything a parameter, it would be barely possible to figure out what’s happening. This is why a good rule of thumb is to use constant values and only add parameters when the function is called with different values at the runtime. In this particular case the query will always contain `configuration.did`, so there’s no reason to make it a parameter. 

And I will say again: "can potentially be used elsewhere" is poison. There’s a rule against it, called the YAGNI principle: "you aint gonna need it".

Reply:
Here a version with your suggestion:

`stash@{0}: On xw/use_indexer: refactor: let the Indexer figure it out`

Reply:
I guess you are trying to attach a file but I don’t have access to it :shrug:

Reply:
I erroneously thought that the stashed commits where also available remotely.

Reply:
I’ve pushed to a new branch [the version of code I’d be happy with](https://github.com/BTE-Trusted-Entity/socialkyc.io/commit/0e884f0af94a81a9d6bfcf858e48a7eeb2b8f6cc)

Reply:
Is there any possibility for the DID to be loaded a bit to late? 

If not, then I like your approach.

Reply:
I will use that.

Reply:
The DID is not loaded from blockchain, this is a configuration parameter, you give it to the application when you start it. The only way it could be pending is when you try to run SKYC on a new blockchain and the DID is not created there yet. In this case SKYC creates the DID and exits, so afterwards you have to add the DID and restart the SKYC.

Reply:
This commit should do the trick:  [06f71ff](https://github.com/BTE-Trusted-Entity/socialkyc.io/pull/2211/commits/06f71fff7e9a886882cded9bd69f16a88b897936)

Reply:
Very nice!